src/flag_gems/ops/argmin.py

```diff
@@ -0,0 +1,178 @@
+    inp,
+    mid_value,
+    mid_index,
+    M,
+    BLOCK_SIZE: tl.constexpr,
+):
+    pid = tle.program_id(0)
+    offset = pid * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
+    inp_ptrs = inp + offset
+    mask = offset < M
+    inp_val = tl.load(inp_ptrs, mask=mask, other=float("inf"))
+    min_val, min_index = tl.min(inp_val, axis=0, return_indices=True)
+    min_index = min_index + pid * BLOCK_SIZE
+    mid_value_ptr = mid_value + pid
+    min_index_ptr = mid_index + pid
+    tl.store(mid_value_ptr, min_val)
+    tl.store(min_index_ptr, min_index)
+
+
+@libentry()
+@triton.jit
+def argmin_kernel_2(mid_value, mid_index, out, mid_size, BLOCK_MID: tl.constexpr):
+    offset = tl.arange(0, BLOCK_MID)
+    mid_ptrs = mid_value + offset
+    mask = offset < mid_size
+    mid_val = tl.load(mid_ptrs, mask=mask, other=float("inf"))
+    index_val = tl.argmin(mid_val, axis=0)
+    mid_index_ptrs = mid_index + index_val
+    out_val = tl.load(mid_index_ptrs)
+    tl.store(out, out_val)
+
+
+def heur_block_n(args):
+    return min(4096, triton.next_power_of_2(args["N"]))
+
+
+@libentry()
+@triton.autotune(
+    configs=runtime.get_triton_config("argmin"),
+    key=[
+        "M",
+        "N",
+    ],
+)
+@triton.heuristics(
+    {
+        "BLOCK_N": heur_block_n,
+    }
+)
+@triton.jit
+def argmin_kernel(
+    inp,
+    out_index,
+    M,
+    N,
+    K,
+    tl_dtype: tl.constexpr,
+    dtype_max_value: tl.constexpr,
+    BLOCK_M: tl.constexpr,
+    BLOCK_N: tl.constexpr,
+):
+    # set offset
+    pid_m = tle.program_id(0)
+    pid_k = tle.program_id(1)
+    m_offset = pid_m * BLOCK_M + tl.arange(0, BLOCK_M)
+
+    # min_values = tl.full([BLOCK_M], dtype=tl.float32, value=float("inf"))
+    min_values = tl.full([BLOCK_M], dtype=tl_dtype, value=dtype_max_value)
+    argmin_values = tl.full([BLOCK_M], dtype=tl.int64, value=0)
+    for start_n in range(0, N, BLOCK_N):
+        n_offset = start_n + tl.arange(0, BLOCK_N)
+        offset = m_offset[:, None] * N * K + n_offset[None, :] * K + pid_k
+        mask = m_offset[:, None] < M and n_offset[None, :] < N
+        inp_ptrs = inp + offset
+        # inp_vals = tl.load(inp_ptrs, mask=mask, other=float("inf"))
+        inp_vals = tl.load(inp_ptrs, mask=mask, other=dtype_max_value)
+        local_min, local_argmin = tl.min(
+            inp_vals, 1, return_indices=True, return_indices_tie_break_left=True
+        )
+        # if return indices is not supported, call a tl.argmin in addition
+        # local_argmin = tl.argmin(inp_vals, 1)
+        update = local_min < min_values
+        min_values = tl.where(update, local_min, min_values)
+        argmin_values = tl.where(update, start_n + local_argmin, argmin_values)
+
+    offset_index = m_offset * K + pid_k
+    out_index_ptrs = out_index + offset_index
+    mask1 = m_offset < M
+    tl.store(out_index_ptrs, argmin_values, mask=mask1)
+
+
+def argmin(inp, dim=None, keepdim=False, *, dtype=None):
+    logging.debug("GEMS argmin")
+    if dim is None:
```

Comment:
Please add test cases where `dim` is `None`. This branch is not tested.

Reply:
DONE